src/reanimated2/hook/useAnimatedStyle.ts

```diff
@@ -510,20 +512,27 @@ For more, see the docs: \`https://docs.swmansion.com/react-native-reanimated/doc
     return () => {
       stopMapper(mapperId);
     };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
   }, dependencies);
 
   useEffect(() => {
-    animationsActive.value = true;
+    areAnimationsActive.value = true;
     return () => {
-      animationsActive.value = false;
+      areAnimationsActive.value = false;
     };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
I think we should add `areAnimationsActive` to the dependencies array and also ensure that `useSharedValue` returns a stable reference.

Reply:
Doesn't `useSharedValue` return a stable reference now with `useState` implementation?

Reply:
Relevant test was added to the PR description.